kayenta-core/src/main/java/com/netflix/kayenta/canary/CanaryMetricSetQueryConfig.java

```diff
@@ -24,8 +24,9 @@
 @JsonInclude(JsonInclude.Include.NON_NULL)
 public interface CanaryMetricSetQueryConfig {
 
-  // Optionally defines an explicit filter to be used when composing the query. Takes precedence over customFilterTemplate.
-  default String getCustomFilter() {
+  // Optionally defines a metric-specific query template. Takes precedence over customFilterTemplate.
+  // It is expanded by using the key/value pairs in extendedScopeParams as the variable bindings.
+  default String getCustomInlineTemplate() {
```

Comment:
Should probably run a manual test to ensure that previously-created configs that were persisted with the original `customFilter` attribute don't cause the updated deserialization logic to choke.

Reply:
May make sense to allow for configs with the deprecated attribute to be deserialized and then throw an exception so the config owner knows to update the config.

Reply:
If you move `customFilter` to the base class, you can throw the exception right in `QueryConfigUtils.expandCustomFilter()`. At least this way it will only notify when someone tries to actually use one of those configs.

Reply:
Awesome thanks, updated. Tested to make sure we get the error back in the UI as well:
![pyazzjrqbx2](https://user-images.githubusercontent.com/15936279/49244921-f1cf8b00-f3de-11e8-9662-6fde0ef6486a.png)